### PR TITLE
Feature/traveler avoid sk

### PR DESCRIPTION
**Pull request by logxen**
This feature raises the pathing cost within a range from SKs. This allows civilian creeps to travel safely through SK rooms.

### CODE REVIEW

File: traveler.js [hunk 1]
```diff
@@ -125,6 +125,14 @@ module.exports = function(globalOpts = {}){
                 else {
                     matrix = this.getCreepMatrix(room);
                 }
+                if (TRAVELER_AVOID_SK) {
+                    let parsed = /^[WE]([0-9]+)[NS]([0-9]+)$/.exec(roomName);
+                    let isSK = ((parsed[1] % 10 === 4) || (parsed[1] % 10 === 6)) &&
+                        ((parsed[2] % 10 === 4) || (parsed[2] % 10 === 6));
+                    if (isSK) {
```
Comment (Spedwards) [inline traveler.js]: Use `Room.isSKRoom(roomName)` instead

Comment (karlthepagan) [inline traveler.js]: References to OCS classes from inside traveler.js will hurt maintainability. Vetoing this change.

**Comment by karlthepagan** [comment]
I also want to avoid changes to traveler.js. Please use the options callback to achieve this feature.

File: traveler.js (hunk 1 above)
Comment (Dasku) [inline traveler.js]: use isCenterNine not is SK room, as SK rooms is only 8, ignoring the center room (which doesnt have an SK)

Comment (Dasku) [inline traveler.js]: use isCenterNine not is SK room, as SK rooms is only 8, ignoring the center room (which doesnt have an SK)

**Comment by chobobobo** [comment]
I have a remote room that transits an SK room, I will merge this in and give it a test today.

**Comment by logxen** [comment]
Great thanks @chobobobo 

@karlthepagan I'll address the other compatibility issues once it is actually working. I mostly pushed this to get help with it.

**Comment by karlthepagan** [comment]
just remove the on hold label when it's ready for canary

**Comment by chobobobo** [comment]
With the changes to traveler, this should be fairly easy to integrate into our existing cost matrices now.

**Comment by karlthepagan** [comment]
I've integrated some of these ideas into a generalized avoiding costmatrix.

**Pull request closed without merging**